### ISSUE TITLE
The nullness checker should not allow @NonNull for uninitialized array elements

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#322]
```
When I run the nullness checker on a class with the following method, the 
checker doesn't report any errors:

  static Object[] createObjects() {
    @NonNull Object [] oa = new @NonNull Object[10];
    return oa;
  }

However, as explained in section 3.3.4 "Nullness and arrays" of the checker 
framework manual, the nullness checker should report an error because the 
elements of the "oa" array are null.

What version of the product are you using? On what operating system?
checker framework 1.8.0, linux

```

Original issue reported on code.google.com by `reprogra...@gmail.com` on 26 Apr 2014 at 2:48

**Comment by mernst** [comment]
This is now properly documented.  However, the default behavior should be sound checking.  I will add a different issue for that.
